docs/import/add_disaster.js

```diff
@@ -60,14 +63,152 @@ function enableWhenReady() {
       });
 }
 
+const layerTypeStrings = new Map();
+for (const t in LayerType) {
+  if (LayerType.hasOwnProperty(t)) {
+    layerTypeStrings.set(LayerType[t], t);
+  }
+}
+
+/**
+ * Write the current state of {@code disasterData} to firestore.
+ * @return {Promise<void>} Returns when finished writing.
+ */
+function writeDataToFirestore() {
```

Comment:
I think we need a queueing mechanism here similar to the one in StoredShapeData#update. Right now, every change kicks off an independent Firestore write. That means that the writes can race, so newer data could be overwritten by older data. The way to fix this (at least, the way it's done in StoredShapeData#update) is to keep track of the current state, Either SAVED, WRITING, or QUEUED_WRITE. Then when a write comes in, you either write it immediately, transitioning to WRITING state (if in SAVED state), do nothing (if in QUEUED_WRITE mode), or go from WRITING to QEUED_WRITE mode but do nothing else.

Then, when a write finishes, if that thread sees that we're in QUEUED_WRITE mode, it changes the mode back to WRITING and writes again. This ensures that we only have one write in flight at any given time, but will write new data as soon as it's safe to do so.